### ISSUE TITLE
Remove lombok slf4j from project

### ISSUE THREAD

**Issue by berndruecker** [camunda/camunda-bpm-spring-boot-starter#152]
Lombok requires own Eclipse plugins to work correctly. The benefit of using
https://projectlombok.org/api/lombok/extern/slf4j/Slf4j.html is minimal - but the effect are puzzling errors in Eclipse. The dependency shall be completly removed.

**Comment by jangalinski** [comment]
We decided to use lombok  for Pojos (configuration) a while ago, see #123.  Slf4j itself is just a side effect. Is this causing you problems when working on this extension or when using it?

**Comment by berndruecker** [comment]
It does - as you have to know about it and install some eclipse plugins to work with it properly (actually it took me half an our to figure out why my eclipse versions does not compile). I think this is not a good idea for a community extensions, which lives from being easy to checkout, build and understand!

**Comment by berndruecker** [comment]
PS: As a framework it is always a better idea to use IDE to generate boring stuff that magic bytecode enhancement. Would never make it to Camunda core as well.

**Comment by jangalinski** [comment]
Any other opinions? @osteinhauer @hawky-4s- ? I really like lombok and want to see it in use more often, but of course I know the troubles you get when you use a non-standard compiler ... 

I see two options: remove lombok, back to 300 lines of getter/setter pairs for our configuration classes or adding a "we use lombok" documentation and how to deal with it to the contrib description ...

**Comment by FrVaBe** [comment]
Beside lombok which uses bytecode instrumentation (as far as I know) there are other projects, which just use code generation (you can see real code after running these tools).

- [Immutabels](https://immutables.github.io/)
- [Google AutoValue](https://github.com/google/auto/tree/master/value)

If it is just about Pojos and Getters/Setters this might be an conceivable alternative approach. 
Nevertheless each library needs a certain level of attention and knowledge. In doubt I would rather agree with @berndruecker to avoid the usage (especially when it only affects small parts of the project).

**Comment by meyerdan** [comment]
You can also just write the damn thing by hand :)

**Comment by jangalinski** [comment]
Thats so 2010

**Comment by jangalinski** [comment]
@FrVaBe  the alternatives you mentioned all rely on AnnotationProcessing. They can only generate new classes, not modify existing. This is not compatible with the spring properties approach. Its eithe rlombok are nothing.

**Comment by berndruecker** [comment]
I still vote for writing it by hand (which can be automated by IDE code template features). Everything else is too much complexity for no value.

**Comment by jangalinski** [comment]
Lets just accept the PR that removes lombok and stop arguing. After all, it's x-mas :-)

**Comment by berndruecker** [comment]
And Santa Claus will deliver the PR? Great :-)

**Comment by FrVaBe** [comment]
I consider to help here. Just to be sure - are you talking about the general removement of lombok from the project (which I supsect and which will affect most of the `org.camunda.bpm.spring.boot.strater.property classes`) or only the @Slf4j part?

I am using eclipse and at least the code of the generated equals/hashCode methods is not very nice. I would probably polish it by hand but in this case it should be unit tested. Thus my question: Do you really need equals/hashCode methods for the property classes?

**Comment by jangalinski** [comment]
Hi Franz ... I would say: remove completely ... so replace the rare slf4j stuff with loggerFactory and let the IDE generate getters and setters for the property classes. That should be it. For equals/hashcode/toString I would create some AbstractProperty class that implements those three methods using commons-lang Builders using reflection. So at least that boilderplate code is hidden.

**Comment by FrVaBe** [comment]
One thing about my pull request: I finally did not implement `equals`/`hashCode` for the property classes because I did not want to deal with the security manager limitations when using the commons lang reflection based builders (I have no knowledge here). If it is necessary to have these methods on the property classes I can have a deeper look. Nevertheless I generated `toString` methods because I think these are very useful for debugging.
